sql/wsrep_sst.cc

```diff
@@ -71,11 +71,14 @@ extern const char *wsrep_defaults_group_suffix;
 #define WSREP_SST_RSYNC "rsync"
 #define WSREP_SST_MYSQLDUMP "mysqldump"
 #define WSREP_SST_SKIP "skip"
+#define WSREP_SST_ONLY_IST "ist_only"
 #define WSREP_SST_XTRABACKUP "xtrabackup"
 #define WSREP_SST_XTRABACKUP_V2 "xtrabackup-v2"
 #define WSREP_SST_DEFAULT WSREP_SST_XTRABACKUP_V2
 #define WSREP_SST_ADDRESS_AUTO "AUTO"
 
+#define WSREP_STATE_TRANSFER_NO_SST ""
```

Comment:
perhaps it makes more sense to declare it at wsrep_api.h where WSREP_STATE_TRANSFER_TRIVIAL and WSREP_STATE_TRANSFER_NONE are also declared.

Also, I believe we should use the define on the wsrep API code where we take action to not send SST request if the return from wsrep_sst_prepare is empty. 

Example, for WSREP_STATE_TRANSFER_TRIVIAL I see that on percona-xtradb-cluster-galera/galera/src/replicator.cpp we populate `Replicator::TRIVIAL_SST(WSREP_STATE_TRANSFER_TRIVIAL);`  we set TRIVIAL_SST variable, and that is used to validate if we are in trivial SST mode. Same should be done for WSREP_STATE_TRANSFER_NO_SST

And possibly the name `WSREP_STATE_TRANSFER_NO_SST` is clashing with Replicator::NO_SST(WSREP_STATE_TRANSFER_NONE);

Reply:
`WSREP_STATE_TRANSFER_TRIVIAL` and `wsrep_api.h` are 3rd party. `WSREP_STATE_TRANSFER_NO_SST` is our thing.
Please note that the functionality which allows us to omit SST, but still have IST is already there in Galera, It is triggered when `sst_request_cb` responsible for preparing SST request string returns an empty string. 
WSREP_STATE_TRANSFER_TRIVIAL - this tells the donor to skip sending SST and IST, but joiner is waiting for IST anyway and gets stuck in JOINED state. Actually, this is what the comment for this define says, but actually it doesn't make much sense.
WSREP_STATE_TRANSFER_NONE - The same as trivial, but according to the comment, the node is supposed to join the cluster. But again it ends up in JONER state waiting for IST.
In fact, there is a mess around these variables. Please check `ReplicatorSMM::process_state_req()` (donor) and `ReplicatorSMM::request_state_transfer` (joiner). SST logic is mixed with IST logic...
So no `Replicator::TRIVIAL_SST`-like things are needed.

Reply:
> Please note that the functionality which allows us to omit SST, but still have IST is already there in Galera, It is triggered when sst_request_cb responsible for preparing SST request string returns an empty string.

That was my point. To have a clear code and make it easier to maintain (in 6 months when you forget the logic of empty string means no SST)  instead of checking for empty string on this code, we should check for a define (this define will translate to empty string) in which we can track back to  WSREP_STATE_TRANSFER_NO_SST, rather than trying to find in the source code all the places where we return empty string.

If that does not make the code any easier to maintain/understand I am ok with accepting the way it is.